python/ray/serve/pipeline/deployment_function_node.py

```diff
@@ -30,19 +32,32 @@ def __init__(
             other_args_to_resolve=other_args_to_resolve,
         )
 
-        if "deployment_self" in self._bound_other_args_to_resolve:
-            original_deployment: Deployment = self._bound_other_args_to_resolve[
-                "deployment_self"
+        if "deployment_schema" in self._bound_other_args_to_resolve:
+            deployment_schema: DeploymentSchema = self._bound_other_args_to_resolve[
+                "deployment_schema"
             ]
-            self._deployment = original_deployment.options(
-                name=(
-                    deployment_name
-                    if original_deployment._name
-                    == original_deployment.func_or_class.__name__
-                    else original_deployment._name
-                ),
-                init_args=tuple(),
+            deployment_shell = schema_to_deployment(deployment_schema)
+
+            # Prefer user specified name to override the generated one.
+            if (
+                inspect.isfunction(func_body)
+                and deployment_shell._name != func_body.__name__
+            ):
+                self._deployment_name = deployment_shell._name
+
+            # Set the route prefix, prefer the one user supplied,
+            # otherwise set it to /deployment_name
+            if deployment_shell._route_prefix != f"/{deployment_shell._name}":
```

Comment:
_route_prefix is a public property that we can directly access via `deployment_shell.route_prefix`